### DISCUSSION TITLE
Flat Scala file outline and filter this outline by class names & function names

### DISCUSSION THREAD

**Discussion by kostafey** [scalameta/metals#5198] [General]
Is there some way to provide flat Scala file outline and filter this outline by class names & function names.
That is, do not show function internals in outline data.
It's ok for VS Code, but very unhandy in Emacs imenu.

Related issue in Emacs lsp-metals: https://github.com/emacs-lsp/lsp-metals/issues/95

**Comment by tgodzik** [comment]
Thanks for the question! Do I understand correctly and this is a problem with documentSymbols? You would want to limit the depth of outline being shown? 

If so I am afraid there is no way to do it inside LSP and I would rather depend on the client to fix the representation. This would need to be very hacky in metals, while this is UI related, so ideally the editor knows best here.
